src/dmd/dinterpret.d

```diff
@@ -454,270 +454,227 @@ struct CompiledCtfeFunction
+            }
         }
-        if (s._body)
-            ctfeCompile(s._body);
-    }
 
-    override void visit(TryCatchStatement s)
-    {
-        debug (LOGCOMPILE)
-        {
-            printf("%s TryCatchStatement::ctfeCompile\n", s.loc.toChars());
-        }
-        if (s._body)
-            ctfeCompile(s._body);
-        foreach (ca; *s.catches)
+        void visitCompoundDeclaration(CompoundDeclarationStatement s)
         {
-            if (ca.var)
-                ccf.onDeclaration(ca.var);
-            if (ca.handler)
-                ctfeCompile(ca.handler);
+            visitCompound(s);
         }
-    }
 
-    override void visit(ThrowStatement s)
-    {
-        debug (LOGCOMPILE)
+        void visitUnrolledLoop(UnrolledLoopStatement s)
         {
-            printf("%s ThrowStatement::ctfeCompile\n", s.loc.toChars());
+            foreach (s2; *s.statements)
+            {
+                if (s2)
+                    visit(s2);
+            }
         }
-        ccf.onExpression(s.exp);
-    }
 
-    override void visit(GotoStatement s)
-    {
-        debug (LOGCOMPILE)
+        void visitScope(ScopeStatement s)
         {
-            printf("%s GotoStatement::ctfeCompile\n", s.loc.toChars());
+            if (s.statement)
+                visit(s.statement);
         }
-    }
 
-    override void visit(ImportStatement s)
-    {
-        debug (LOGCOMPILE)
+        void visitDefault(DefaultStatement s)
         {
-            printf("%s ImportStatement::ctfeCompile\n", s.loc.toChars());
+            if (s.statement)
+                visit(s.statement);
         }
-        // Contains no variables or executable code
-    }
 
-    override void visit(ForeachRangeStatement s)
-    {
-        debug (LOGCOMPILE)
+        void visitWith(WithStatement s)
         {
-            printf("%s ForeachRangeStatement::ctfeCompile\n", s.loc.toChars());
+            // If it is with(Enum) {...}, just execute the body.
+            if (s.exp.op == TOK.scope_ || s.exp.op == TOK.type)
+            {
+            }
+            else
+            {
+                dgVar(s.wthis);
+                dgExp(s.exp);
+            }
+            if (s._body)
+                visit(s._body);
         }
-        // rewritten for ForStatement
-        assert(0);
-    }
 
-    override void visit(AsmStatement s)
-    {
-        debug (LOGCOMPILE)
+        void visitTryCatch(TryCatchStatement s)
         {
-            printf("%s AsmStatement::ctfeCompile\n", s.loc.toChars());
-        }
-        // we can't compile asm statements
-    }
-
-    void ctfeCompile(Statement s)
-    {
-        s.accept(this);
-    }
+            if (s._body)
+                visit(s._body);
+            foreach (ca; *s.catches)
+            {
+                if (ca.var)
+                    dgVar(ca.var);
+                if (ca.handler)
+                    visit(ca.handler);
+            }
+        }
+
+        void visitTryFinally(TryFinallyStatement s)
```

Comment:
Aren't the names of these functions redundant?  Can't they all be named `visit` and have D's overload rules handle the resolution?

Reply:
Overloading is not supported for nested functions.